maven-plugin-tools-api/src/main/java/org/apache/maven/tools/plugin/extractor/GroupKey.java

```diff
@@ -0,0 +1,121 @@
+package org.apache.maven.tools.plugin.extractor;
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import java.util.Objects;
+
+/**
+ * Group key: defines "grouping" for descriptor (based on source of extraction) and rank within
+ * group.
+ *
+ * @since TBD
+ */
+public final class GroupKey
+    implements Comparable<GroupKey>
+{
+    /**
+     * Java group is handled a bit special: is always first to be scanned.
+     */
+    public static final String JAVA_GROUP = "java";
+
+    private final String group;
+
+    private final int order;
+
+    public GroupKey( String group, int order )
+    {
+        if ( group == null )
+        {
+            throw new NullPointerException( "GroupKey.group null" );
```

Comment:
`Objects#requireNonNull`?

Reply:
This is Java7 project, but yes, we may want to up it to Java8 for 3.7.0